packages/flutter_tools/lib/src/commands/apk.dart

```diff
@@ -0,0 +1,201 @@
+
+  ApkCommand() {
+    argParser.addOption('manifest',
+        abbr: 'm',
+        defaultsTo: _kDefaultAndroidManifestPath,
+        help: 'Android manifest XML file.');
+    argParser.addOption('output-file',
+        abbr: 'o',
+        defaultsTo: _kDefaultOutputPath,
+        help: 'Output APK file.');
+    argParser.addOption('target',
+        abbr: 't',
+        defaultsTo: '',
+        help: 'Target app path or filename used to build the FLX..');
+    argParser.addOption('flx',
+        abbr: 'f',
+        defaultsTo: '',
+        help: 'Path to the FLX file. If this is not provided, an FLX will be built.');
+  }
+
+  void _buildApk(String flxPath) {
+    File androidManifest = new File(argResults['manifest']);
+    File icuData = new File('${runner.enginePath}/third_party/icu/android/icudtl.dat');
+    File classesDex = new File('${runner.enginePath}/out/android_Debug/gen/sky/shell/shell/classes.dex');
+    File libSkyShell = new File('${runner.enginePath}/out/android_Debug/gen/sky/shell/shell/shell/libs/armeabi-v7a/libsky_shell.so');
+    File keystore = new File('${runner.enginePath}/build/android/ant/chromium-debug.keystore');
+
+    for (File f in [androidManifest, icuData, classesDex, libSkyShell, keystore]) {
+      if (!f.existsSync()) {
+        _logging.severe('Can not locate file: ${f.path}');
+        return;
+      }
+    }
+
+    Directory androidSdk = new Directory('${runner.enginePath}/third_party/android_tools/sdk');
+    if (!androidSdk.existsSync()) {
+      _logging.severe('Can not locate Android SDK: ${androidSdk.path}');
+      return;
+    }
+
+    Directory tempDir = Directory.systemTemp.createTempSync('flutter_tools');
+    try {      
+      AssetBuilder assetBuilder = new AssetBuilder(tempDir, 'assets');
+      assetBuilder.add(icuData, 'icudtl.dat');
+      assetBuilder.add(new File(flxPath), 'app.flx');
+
+      AssetBuilder artifactBuilder = new AssetBuilder(tempDir, 'artifacts');
+      artifactBuilder.add(classesDex, 'classes.dex');
+      artifactBuilder.add(libSkyShell, 'lib/armeabi-v7a/libsky_shell.so');
+
+      ApkBuilder builder = new ApkBuilder(androidSdk.path);
+      File unalignedApk = new File('${tempDir.path}/app.apk.unaligned');
+      builder.package(unalignedApk, androidManifest, assetBuilder.directory,
+                      artifactBuilder.directory);
+      builder.sign(keystore, _kKeystorePassword, _kKeystoreKeyName, unalignedApk);
+
+      File finalApk = new File(argResults['output-file']);
+      builder.align(unalignedApk, finalApk);
+    } finally {
+      tempDir.deleteSync(recursive: true);
+    }
+  }
+
+  @override
+  Future<int> runInProject() async {
+    if (runner.enginePath == null) {
+      _logging.severe('Unable to locate the Flutter engine.  Use the --engine-src-path option.');
+      return;
+    }
+
+    String flxPath = argResults['flx'];
+
+    if (!flxPath.isEmpty) {
+      if (!(new File(flxPath).existsSync())) {
```

Comment:
FileSystemEntity.isFileSync(flxPath) will let you do this more directly.  (Also will test that its a file and not a directory, which won't work later.)